meson.build

```diff
@@ -514,6 +515,10 @@ if not posix and not features['win32-desktop']
                      'osdep/terminal-dummy.c')
 endif
 
+if not features['cocoa']
+    sources += files('osdep/clipboard-dummy.c')
```

Comment:
The "one global handler" approach will not work on Linux. Both the X11 and Wayland code needs to be compiled in, but only one of them will be active depending on the VO.

Reply:
I'd expect to have a single linux implementation of the `m_clipboard_*` functions that dispatches between X and Wayland as needed.

Reply:
Requiring only one implementation per platform sounds unnecessarily inflexible to me and as you proposed also leads to weird patterns where you have code that only forwards calls. Plus not being able to handle edge cases like X11 on macOS (just an example, not suggesting we should have that).

Reply:
There can be multiple implementations of the underlying functionality per platform, they just need a implementation of the cross-platform APIs; if e.g. `clipboard-linux.c` wants to call into specific routines in `clipboard-wayland.c` and `clipboard-x11.c`, that's fine.